modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -103,11 +103,36 @@ object Run extends ScalaCommand[RunOptions] with BuildCommandHelpers {
   }
 
   def run(
-    options: RunOptions,
+    opts: RunOptions,
     inputArgs: Seq[String],
     programArgs: Seq[String],
     defaultInputs: () => Option[Inputs]
   ): Unit = {
+    val options =
+      if (!opts.sharedRun.sharedDebug.debug)
+        opts
+      else
+        val server = opts.sharedRun.sharedDebug.debugMode match {
```

Comment:
If you move debug options to `SharedJvmOptions`, and then resolve debug java opts in `JvmUtils`. Support `--debug` flag for `test` command is for free.